context/src/main/java/io/grpc/Deadline.java

```diff
@@ -52,24 +51,25 @@
    * @return A new deadline.
    */
   public static Deadline after(long duration, TimeUnit units) {
-    return after(duration, units, SYSTEM_TICKER);
+    checkNotNull(units, "units");
+    return new Deadline(SystemTicker.INTANCE, units.toNanos(duration), true);
   }
 
-  // For testing
-  static Deadline after(long duration, TimeUnit units, Ticker ticker) {
+  //For testing
+  public static Deadline forTest(long duration, TimeUnit units, DeadlineTicker ticker) {
```

Comment:
It's technically impossible to stop people from using it in production for whatever reason. It is mimicking the guava Ticker which is not test-only, so I don't see a reason this has to be test-only.